### PR TITLE
1) Fix issue that breakpoints were only matched against bkptno result…

### PR DESCRIPTION
**Pull request by jacdavis**
…ing in bugs when multple bps bind to the same place such as the entrypoint bp. 2) Use the trap command to setup a signal that cleans up the fifos in the local linux case if the user closes the debuggee terminal

### CODE REVIEW

File: src/MIDebugEngine/Engine.Impl/BreakpointManager.cs [hunk 1]
```diff
@@ -141,29 +141,66 @@ private AD7PendingBreakpoint BindToAddress(string bkptno, ulong addr, /*OPTIONAL
...
-        public AD7BoundBreakpoint FindHitBreakpoint(string bkptno, ulong addr, /*OPTIONAL*/ TupleValue frame, out bool fContinue)
+        // Return all bound breakpoints bound to an address regardless of bptno
+        AD7BoundBreakpoint[] FindBoundBreakpointsAtAddress(ulong addr)
+        {
+            lock (_pendingBreakpoints)
+            {
+                List<AD7BoundBreakpoint> matchingBoundBreakpoints = new List<AD7BoundBreakpoint>();
+                foreach (AD7PendingBreakpoint currPending in _pendingBreakpoints)
+                {
+                    matchingBoundBreakpoints.AddRange(Array.FindAll(currPending.EnumBoundBreakpoints(), (b) => b.Addr == addr));
```
Comment (gregg-miskelly) [inline src/MIDebugEngine/Engine.Impl/BreakpointManager.cs]: Do we need to exclude Disabled/Deleted/PendingDeleted breakpoints?

File: src/MIDebugEngine/Engine.Impl/BreakpointManager.cs [hunk 2]
```diff
@@ -141,29 +141,66 @@ private AD7PendingBreakpoint BindToAddress(string bkptno, ulong addr, /*OPTIONAL
...
+                // However, clrdbg does not support addresses on its breakpoints, so the bkptno code path is
+                // preserved.
+                if (addr != 0)
                 {
-                    return null;
+                    hitBps = FindBoundBreakpointsAtAddress(addr);
                 }
-                fContinue = true;
-                if (!pending.Enabled || pending.Deleted || pending.PendingDelete)
+                else
```
Comment (gregg-miskelly) [inline src/MIDebugEngine/Engine.Impl/BreakpointManager.cs]: Do we need to lookup by breakpoint number in the case that we previously didn't know the address of the breakpoint? I think that can happen if the breakpoint binds to multiple addresses.

**Comment by gregg-miskelly** [comment]
Signing off with comments.

**Comment by paulmaybee** [comment]
:+1:

**Comment by gregg-miskelly** [comment]
:+1:

**Pull request merged**